Fix parsing copy constructor

Adda0 opened the pull request:
This PR optimizes parsing by introducing move assignment operator and move constructor for `FormulaGraph` and requiring some moves to happen during the parsing. It is just a quick improvement for now. Modifications to the overall algorithms are needed which will probably completely rewrite these functions.

@martinhruska Feel free to give me your input on these changes. As discussed privately, this helps, but is not enough.

Adda0 committed:
Stop calling copy constructor repeatedly

Adda0 committed:
Use deque to allow efficient insert to the front

Adda0 committed:
Optimize working with intermediate automaton

Adda0 committed:
Remove unnecessary consts

Adda0 committed:
Optimize code using moves and views

Adda0 commented:
I will have a look what the MacOS error is tomorrow.

Adda0 committed:
Add move constructor

Adda0 committed:
Refactor to never push to front

Adda0 committed:
Comment out Clang unsupported std::ranges::all_of() function

tfiedor commented on src/inter-aut.cc:
```diff
@@ -308,7 +309,7 @@ namespace {
      * @return A postfix with eventually added operators
      */
-    std::vector<mata::FormulaNode> add_disjunction_implicitly(const std::vector<mata::FormulaNode> &postfix)
+    std::vector<mata::FormulaNode> add_disjunction_implicitly(std::vector<mata::FormulaNode> postfix)
```
Why did you remove the `&` reference? Isn't this less efficient?

Adda0 replied:
Because we are moving the data from the caller to the function parameter. Hence, we can afterwards move the elements of the postfix vector into the output vector. Otherwise, we would have to create a new copy in the output vector.

tfiedor reviewed (approved):
Seems ok

Adda0 commented:
Due to some new surprising discoveries, I ask the reviews to wait with their reviews. Nothing much should change, but there might be some new modifications introduced.

Adda0 commented:
The changes since the last time have great impact on the performance, but are minuscule in the code itself. The parser should be about 10.5 times faster on average than the current parser. On the examples I tried, we get from 3.8 s to 0.36 s. It might still be too slow for our needs, but it is definitely better. At least, it is a first step.   

@jurajsic Could you please look at the PR and review the changes now?
